new-log-viewer/src/components/Editor/MonacoInstance/index.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+    useEffect,
+    useRef,
+} from "react";
+
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+
+import {ActionType} from "../../../utils/actions";
+import {
+    CursorExplicitPosChangeCallback,
+    CustomActionCallback,
+} from "./typings";
+import {
+    goToPositionAndCenter,
+    initMonacoEditor,
+} from "./utils";
+
+import "./index.css";
+
+
+interface MonacoEditorProps {
+    lineNum: number,
+    text: string,
+    actions: ActionType[],
+    beforeMount?: () => void,
+    beforeTextUpdate?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+    onCursorExplicitPosChange: CursorExplicitPosChangeCallback,
+    onCustomAction: CustomActionCallback,
+    onMount?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+    onTextUpdate?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+}
+
+/**
+ * Wraps a `monaco-editor` instance created from the DOM rendered, which accepts a variety of props
+ * to configure text content, custom actions, and various lifecycle hooks for interacting with
+ * the editor.
+ *
+ * @param props
+ * @param props.text
+ * @param props.lineNum
+ * @param props.actions
+ * @param props.beforeMount
+ * @param props.beforeTextUpdate
+ * @param props.onCursorExplicitPosChange
+ * @param props.onCustomAction
+ * @param props.onMount
```

Comment:
Can we document the args that `beforeTextUpdate` and `onMount` take / would it be hard to add a typing for these args so that we don't need to document the args?

Reply:
Let's add the typings and we can add docs for the typings.